base/channels.jl

```diff
@@ -37,6 +37,7 @@ mutable struct Channel{T} <: AbstractChannel{T}
     excp::Union{Exception, Nothing}      # exception to be thrown when state !== :open
 
     data::Vector{T}
+    @atomic length::Int                  # Length, tracked atomically
```

Comment:
Given the discussion in the thread https://github.com/JuliaLang/julia/pull/41833#discussion_r694224135 maybe it's better to rename this to `n_avail`?